File: src/JFrameDemo1.java
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.SwingUtilities;

public class JFrameDemo1 {
    private static class MyRunnable implements Runnable{
        @Override
        public void run() {
            constructGUI();
        }
    }

    private static void constructGUI(){
        JFrame frame = new JFrame();
        frame.setTitle("My First Swing Application");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        JLabel label = new JLabel("Welcome");
        frame.add(label);
        frame.pack();
        frame.setVisible(true);
    }
    public static void main(String[] args){
        MyRunnable doSomething = new MyRunnable();
        SwingUtilities.invokeLater(doSomething);
    }
    /*
    public static void main(String[] args){
        SwingUtilities.invokeLater(new Runnable() {
            @Override
            public void run() {
                constructGUI();
            }
        });
    }
    */



}

File: src/JFrameDemo1.kt
import javax.swing.JFrame
import javax.swing.JLabel
import javax.swing.SwingUtilities

fun constructGUI(){
    val frame = JFrame()
    frame.title = "My First Swing Application"
    frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
    val label = JLabel("welcome")
    frame.add(label)
    frame.pack()
    frame.isVisible = true
}

fun main(){
    //SwingUtilities.invokeLater(::constructGUI)
    /*
    SwingUtilities.invokeLater(Runnable {
        constructGUI()
    })
    */
     SwingUtilities.invokeLater {
         constructGUI()
     }
}